tests/net/socket/tcp/src/main.c

```diff
@@ -446,8 +446,73 @@ void test_v4_accept_timeout(void)
 	k_sleep(TCP_TEARDOWN_TIMEOUT);
 }
 
+#ifdef CONFIG_USERSPACE
```

Comment:
The tests look good. I wish we had these originally so would not have missed the issue.
We probably also need to add tests to other socket types like net_mgmt, websocket etc.

Reply:
Agreed, I think the end goal we should strive for is to exercise the cross product of all socket APIs against all socket types.